engine/common/follower/cache/cache_test.go

```diff
@@ -105,101 +110,102 @@ func (s *CacheSuite) TestBlocksAreNotConnected() {
+	_, err := s.cache.AddBlocks(blocks[1:2])
 	require.NoError(s.T(), err)
-	certifiedBatch, certifyingQC, err := s.cache.AddBlocks([]*flow.Block{blocks[0]})
+	certifiedBatch, err := s.cache.AddBlocks(blocks[0:1])
 	require.NoError(s.T(), err)
 	require.Len(s.T(), certifiedBatch, 1)
-	require.NotNil(s.T(), certifyingQC)
-	require.Equal(s.T(), blocks[0].ID(), certifyingQC.BlockID)
-	require.Equal(s.T(), certifiedBatch[0], blocks[0])
+	require.Equal(s.T(), blocks[0].Block.ID(), certifiedBatch[0].CertifyingQC.BlockID)
+	require.Equal(s.T(), blocks[0], certifiedBatch[0].Proposal)
 }
 
 // TestBlockInTheMiddle tests a scenario: A <- B[QC_A] <- C[QC_B].
 // We add blocks one by one: C, A, B, we expect that after adding B, we will be able to
 // certify [A, B] with QC_B as certifying QC.
 func (s *CacheSuite) TestBlockInTheMiddle() {
-	blocks := unittest.ChainFixtureFrom(3, unittest.BlockHeaderFixture())
+	blocks := unittest.ProposalChainFixtureFrom(3, unittest.BlockHeaderFixture())
 	// add C
-	certifiedBlocks, certifiedQC, err := s.cache.AddBlocks(blocks[2:])
+	certifiedBlocks, err := s.cache.AddBlocks(blocks[2:])
 	require.NoError(s.T(), err)
 	require.Empty(s.T(), certifiedBlocks)
-	require.Nil(s.T(), certifiedQC)
 
 	// add A
-	certifiedBlocks, certifiedQC, err = s.cache.AddBlocks(blocks[:1])
+	certifiedBlocks, err = s.cache.AddBlocks(blocks[:1])
 	require.NoError(s.T(), err)
 	require.Empty(s.T(), certifiedBlocks)
-	require.Nil(s.T(), certifiedQC)
 
 	// add B
-	certifiedBlocks, certifiedQC, err = s.cache.AddBlocks(blocks[1:2])
+	certifiedBlocks, err = s.cache.AddBlocks(blocks[1:2])
 	require.NoError(s.T(), err)
-	require.Equal(s.T(), blocks[:2], certifiedBlocks)
-	require.Equal(s.T(), blocks[2].Header.QuorumCertificate(), certifiedQC)
+	require.Equal(s.T(), blocks[0], certifiedBlocks[0].Proposal)
+	require.Equal(s.T(), blocks[len(blocks)-2], certifiedBlocks[len(certifiedBlocks)-1].Proposal)
+	require.Equal(s.T(), blocks[2].Block.Header.QuorumCertificate(), certifiedBlocks[1].CertifyingQC)
 }
 
 // TestAddBatch tests a scenario: B1 <- ... <- BN added in one batch.
 // We expect that all blocks except the last one will be certified.
 // Certifying QC will be taken from last block.
 func (s *CacheSuite) TestAddBatch() {
-	blocks := unittest.ChainFixtureFrom(10, unittest.BlockHeaderFixture())
-	certifiedBatch, certifyingQC, err := s.cache.AddBlocks(blocks)
+	blocks := unittest.ProposalChainFixtureFrom(10, unittest.BlockHeaderFixture())
+	certifiedBatch, err := s.cache.AddBlocks(blocks)
 	require.NoError(s.T(), err)
-	require.Equal(s.T(), blocks[:len(blocks)-1], certifiedBatch)
-	require.Equal(s.T(), blocks[len(blocks)-1].Header.QuorumCertificate(), certifyingQC)
+	require.Equal(s.T(), blocks[0], certifiedBatch[0].Proposal)
+	require.Equal(s.T(), blocks[len(blocks)-2], certifiedBatch[len(certifiedBatch)-1].Proposal)
+	require.Equal(s.T(), blocks[len(blocks)-1].Block.Header.QuorumCertificate(), certifiedBatch[len(certifiedBatch)-1].CertifyingQC)
```

Comment:
for maximal coverage of the test, I would verify that _each_ block has the matching QC
```suggestion
	require.NoError(s.T(), err)
	require.Len(s.T(), certifiedBatch, 9, "there should be %d - 1 certified blocks", len(blocks))
	for i := 0; i < len(certifiedBatch)-1; i++ {
		certifiedBlock := certifiedBatch[i]
		require.Equal(s.T(), blocks[i], certifiedBlock.Proposal)
		require.Equal(s.T(), blocks[i+1].Block.Header.QuorumCertificate(), certifiedBlock.CertifyingQC)
		require.Equal(s.T(), certifiedBlock.Proposal.Block.ID(), certifiedBlock.CertifyingQC.BlockID)
		require.Equal(s.T(), certifiedBlock.Proposal.Block.Header.View, certifiedBlock.CertifyingQC.View)
	}
```